olp-cpp-sdk-core/include/olp/core/network2/NetworkTypes.h

```diff
@@ -44,5 +44,58 @@ enum class RequestIdConstants : std::uint64_t {
   RequestIdMax = std::numeric_limits<RequestId>::max(),
 };
 
+/**
+ * @brief Common Network error codes.
+ */
+enum class ErrorCode {
+  NO_ERROR = 0,
+  IO_ERROR = -1,
+  AUTHORIZATION_ERROR = -2,
+  INVALID_URL_ERROR = -3,
+  OFFLINE_ERROR = -4,
+  CANCELLED_ERROR = -5,
+  AUTHENTICATION_ERROR = -6,
+  TIMEOUT_ERROR = -7,
+  UNKNOWN_ERROR = -8,
+};
+
+/**
+ * @brief Helper class representing the outcome of making a network request.
+ * It will contain either a valid RequestId or the error code in case request
+ * trigger failed.  The caller must check whether the outcome of the request was
+ * a success before attempting to access the result or the error.
+ */
+class NetworkStatus {
```

Comment:
```suggestion
class SendRequestOutcome {
```
I guess NetworkStatus is a bit misleading name. For me it sounds like the status of network connection - for example, _offline_, _online_. I see that AWS using `Outcome` class as a analogue of our `ApiResponse` class:  https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/error-handling.html

So, I suggest similar name strategy - like "`NetworkRequestOutcome`" or "`SendRequestOutcome`", WDYT?